### PR TITLE
add github action for release and doc building

### PR DESCRIPTION
**Pull request by weikang9009**
Add Github actions for release and docs building - resolve part of the infrastructure issue https://github.com/pysal/pointpats/issues/86

### CODE REVIEW

File: .github/workflows/release_and_publish.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine jupyter urllib3 pandas pyyaml
+          python setup.py sdist bdist_wheel
+
+      - name: Publish distribution 📦 to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_PASSWORD }}
```
Comment (martinfleis) [inline .github/workflows/release_and_publish.yml:44]: Have you added the PyPI token as a repository secret?

**Comment by martinfleis** [comment]
I'll leave the review of the docs action to @jGaboardi but it will probably need some other action as this is essentially a move from RTD to PySAL.org/pointpats. Not sure how that works.

File: .github/workflows/release_and_publish.yml (hunk 1 above)
Comment (weikang9009) [reply .github/workflows/release_and_publish.yml:44]: Yes, just added it.

**Comment by jGaboardi** [comment]
> The workflow needs to be integrated into the repo before configuring the setting of github page. @jGaboardi let me know if my understanding is incorrect.

@weikang9009, yes I believe your understanding is correct (if I remember correctly).

> this is essentially a move from RTD to PySAL.org/pointpats. Not sure how that works.

I also can't remember how this works since it has been some years since I've used RTD. In any case, I think it's a good idea to retain the  [`readthedocs.yml`](https://github.com/pysal/pointpats/blob/master/readthedocs.yml) file for now until things are worked out.

**Comment by jGaboardi** [comment]
Also, @weikang9009 do you have admin privileges here in `pointpats`? We should update the default branch from `master` to `main` after this PR if you do.

**Review by jGaboardi — APPROVED** [review]

**Pull request merged**